src/main/java/model/input/file/PlainTextFile.java

```diff
@@ -127,30 +127,20 @@ private void addRowsToBuilder(List<String> lines) {
 	/**
 	 * Creates a DataValue from a string.
 	 * @param value The string that will be converted
-	 * @param type The type of the column in which the value will be inserted.
+	 * @param columnInfo The info of the column to which the DataValue will be inserted
 	 * @return The DataValue
 	 */
-	private DataValue toDataValue(String value, Class<? extends DataValue> type) {
-		if (type.equals(StringValue.class)) {
-			return new StringValue(value);
-		} else if (type.equals(IntValue.class)) {
-			if (tryParseInt(value)) {
-				return new IntValue(Integer.parseInt(value));
-			} else {
-				// TODO: throw appropriate exception
-				return null;
-			}
-		} else if (type.equals(FloatValue.class)) {
-			if (tryParseFloat(value)) {
-				return new FloatValue(Float.parseFloat(value));
-			} else {
-				// TODO: throw appropriate exception
-				return null;
-			}
+	private DataValue toDataValue(String value, ColumnInfo columnInfo) {
+		if (value.isEmpty()) {
+			return createNullValue(columnInfo.getType());
+		} else if (columnInfo.getType() == DateValue.class) {
```

Comment:
het is raar om hier slechts te controleren op de helft van de types. Je moet of alle type controleren en dan hier per type de juiste value aan maken, of je moet een methode aanroepen die zelf kan bekijken wat hij voor dat type moet doen.